### PR TITLE
Implements accessing the UUIDs of a config items by tag reference

### PR DESCRIPTION
**Pull request by ndejong**
Raised as issue https://github.com/opnsense/core/issues/2664
Discussed in [forum here](https://forum.opnsense.org/index.php?topic=9489.0)

### CODE REVIEW

**Comment by ndejong** [comment]
Even tried creating a new branch to prevent those previous commits getting tangled up with this PR here - oh, git...

File: src/opnsense/service/modules/addons/template_helpers.py [hunk 1]
```diff
@@ -111,6 +111,18 @@ def getUUID(self, uuid):
...
+        """ retrieve a list of uuids from the item list returned by tag
+        :param tag: tag in dot notation (section.item)
+        :return: []
+        """
+        uuids = []
+        for item in self.toList(tag):
+            for uuid in self._template_in_data['__uuid__']:
+                if self._template_in_data['__uuid__'][uuid] == item:
+                    uuids.append(uuid)
+        return uuids
```
Comment (fabianfrz) [inline src/opnsense/service/modules/addons/template_helpers.py]: should be done for a single element

**Comment by ndejong** [comment]
This updated diff makes it more explicit about there the `uuid` string is coming from - also adds a `break` once the `uuid` is found since we don't need to keep searching and can skip to the next

**Comment by AdSchellevis** [comment]
@ndejong if we add the attributes to the object, would that be sufficient for you? I'm not sure we should add an extractor for all uuids if most use-cases need the uuid of the item traversing.

**Comment by ndejong** [comment]
@AdSchellevis  - I'm probably happy with any solution that is clear and easy to use.

If step back for a moment and consider why I wrote an extractor - As a relative newbie in the OPNsense codebase I felt like I wanted a solution that made the templates easier to write and understand.  To get there I first dug through other templates and then through code to discover the data-structures inside the template helpers at which point it did not occur to me that I could simply use the item object itself as the comparator as @fabianfrz pointed out in the Nginx [work around](https://github.com/opnsense/plugins/blob/94a55d5e61668ed13cc4cc245911294d9dadcfc2/www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf#L138-L144) - my extractor uses the same approach.

The idea of injecting a special `__uuid__` attribute into each item certainly works, perhaps the comment I'd make is that, again as a newbie it took me until I discovered where the template helper was and then finding a way to observe the data-structures inside that the `__uuid_tags__` and `__uuid__` elements became apparent - there was all a bit magic  :)  and so I wonder if the per item `__uuid__` attribute will be too much magic and/or get confused with the existing `__uuid__` element that feels like it could have been called `__uuids__` (with the "s")

I'm perhaps biased here, but I feel like the explicit function that does what it says `getUUIDs()` makes it easier for the next plugin developer that comes stumbling along.

Totally your call.

**Comment by fabianfrz** [comment]
@AdSchellevis the uuid as a property would be exactly what I need. be careful because you may overwrite data. In PHP i get it as [(at) attibutes]["uuid"]. The advantage is, that an @ cannot be a valid XML tag.

**Comment by AdSchellevis** [comment]
@fabianfrz I know, As soon as I have a proper fix, I'll post it in the issue https://github.com/opnsense/core/issues/2664 and close this one when done.

**Comment by AdSchellevis** [comment]
@ndejong @fabianfrz this https://github.com/opnsense/core/commit/3add6c7cc9adea860061a8849286047cb388e777 should do the trick. 

To use a uuid, just use something similar to:

```
  {{item['@uuid']}}
```

**Comment by ndejong** [comment]
Most excellent indeed

**Comment by ndejong** [comment]
@AdSchellevis yes please do

**Pull request closed without merging**